spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/DynamicFileFilter.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.spark.sql.catalyst.plans.logical
+
+import org.apache.spark.sql.catalyst.expressions.Attribute
+import org.apache.spark.sql.execution.datasources.v2.DataSourceV2ScanRelation
+
+// TODO: fix stats (ignore the fact it is a binary node and report only scanRelation stats)
+case class DynamicFileFilter(
```

Comment:
@rdblue, stats is something we will have to think about here.

Reply:
Agreed. Can we just delegate to the relation?

Reply:
I don't think so, unfortunately. Nodes themselves don't control this. For example, `LogicalPlanVisitor` and its implementation, `SizeInBytesOnlyStatsPlanVisitor`, has the following logic:

```
  override def default(p: LogicalPlan): Statistics = p match {
    case p: LeafNode => p.computeStats()
    case _: LogicalPlan =>
      Statistics(sizeInBytes = p.children.map(_.stats.sizeInBytes).filter(_ > 0L).product)
  } 
```

Here, we have a binary node so the output will be a product. We could wrap the filter plan into something that reports 1 as its size in bytes. That way, the product will produce the size of the scan relation. Any better ideas?

Reply:
We could separate the file filter query into its own execution that is run in `DynamicFileFilter.doPrepare`. That way it isn't part of this plan.

Reply:
@rdblue, could you elaborate a bit on this one?

Reply:
We could separate the plan out of this one, so that it doesn't show up in this plan, but gets run in `doPrepare`.